src/Products/DetailedProductEntity.php

```diff
@@ -37,15 +43,16 @@ public function getRelated(): int
 
     public function jsonSerialize(): mixed
     {
+        $dimensions = UnitConversionService::unitConverter($_GET['unit'], $this->getDepth(),$this->getWidth(),$this->getHeight());
```

Comment:
Don't use `$_GET['unit']` here, use a setter to set a static property of the unit on the `UnitConversionService` class (ideally within `product.php` or wherever this is used)